agent/src/main/java/com/microsoft/hydralab/agent/runner/ActionExecutor.java

```diff
@@ -29,54 +29,59 @@ public class ActionExecutor {
     /**
      * the implementation of supported actions should not be overload
      */
-    private Set<String> actionTypes = Set.of("setProperty", "setDefaultLauncher", "backToHome", "changeGlobalSetting", "changeSystemSetting");
+    private Set<String> actionTypes = Set.of("setProperty", "setDefaultLauncher", "backToHome", "changeGlobalSetting", "changeSystemSetting", "pushFileToDevice", "pullFileFromDevice");
 
-    public void doActions(@NotNull DeviceManager deviceManager, @NotNull DeviceInfo deviceInfo, @NotNull Logger logger,
-                          @NotNull Map<String, List<DeviceAction>> actions, @NotNull String when) {
+    public boolean doActions(@NotNull DeviceManager deviceManager, @NotNull DeviceInfo deviceInfo, @NotNull Logger logger,
```

Comment:
Maybe we can return an exception or an exception list here, so we can know more detail about the execute status in caller level.